File: 1.5 ariprog/ariprog.java
/*
ID: angela.14
LANG: JAVA
TASK: ariprog
*/
import java.io.*;
import java.util.*;

public class ariprog {
    public static void main(String[] args) throws IOException {
        BufferedReader fin = new BufferedReader(new FileReader("ariprog.in"));
        PrintWriter fout = new PrintWriter(new BufferedWriter(new FileWriter("ariprog.out")));
        int N = Integer.parseInt(fin.readLine());
        int M = Integer.parseInt(fin.readLine());

        int max = M * M * 2;
        boolean[] bisquares = new boolean[max + 1];
        ArrayList<int[]> results = new ArrayList<>();
        for (int p = 0; p <= M; p++) {
            for (int q = 0; q <= M; q++) {
                bisquares[p * p + q * q] = true;
            }
        }

        for (int i = 0; i < max; i++) {
            if (!bisquares[i])
                continue;
            diffLoop: for (int j = 1; j <= (max - i) / (N - 1); j++) {
                for (int k = 1; k <= N - 1; k++) {
                    if (!bisquares[i + j * k])
                        continue diffLoop;
                }
                results.add(new int[] { i, j });
            }
        }

        if (results.size() > 1) {
            Collections.sort(results, new Comparator<int[]>() {
                public int compare(int[] prog1, int[] prog2) {
                    int compTerms = prog1[1] - prog2[1];
                    if (compTerms != 0) {
                        return compTerms;
                    }
                    return prog1[0] - prog2[0];
                }
            });
        }

        if (results.size() > 0) {
            for (int[] result : results) {

                fout.println(result[0] + " " + result[1]);
            }
        } else {
            fout.println("NONE");
        }
        fout.close();
    }

}

File: 1.5 ariprog/ariprog.py
"""
ID: angela.14
LANG: PYTHON3
TASK: ariprog
"""
import math


with open('ariprog.in', 'r') as fin:
    N = int(fin.readline().strip())
    M = int(fin.readline().strip())

bisquares = set()
last_element = 0

for p in range(M+1):
    for q in range(M+1):
        bisquares.add(p*p+q*q)
        if p == M and q == M:
            last_element = p*p+q*q
# print(bisquares)

def check_if_progression(start, difference):
    number_of_terms = 1
    curr = start
    while number_of_terms < N:
        if curr + difference in bisquares:
            curr += difference
            number_of_terms += 1
        else:
            return False
    return True


ari_prog_sequences = []

for val in bisquares:
    for j in range(1, math.floor(last_element / (N-1))+1):
        if check_if_progression(val, j) == True:
            ari_prog_sequences.append((j, val))

ari_prog_sequences = sorted(ari_prog_sequences)

with open('ariprog.out', 'w') as fout:
    length = len(ari_prog_sequences)
    if length ==  0:
        fout.write(f"NONE\n")
    else:
        for i in range(length):
            fout.write(f"{ari_prog_sequences[i][1]} {ari_prog_sequences[i][0]}\n")